src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -324,6 +333,56 @@ def __send_helm_release_events(self, release_data: List[HelmRelease]):
         except Exception:
             logging.error("Error occurred while sending `helm release` trigger event", exc_info=True)
 
+    def __discover_custom_namespaced_resources(self):
+        self.__assert_namespaces_cache_initialized()
+
+        logging.info(f"Starting namespace discovery loop every {NAMESPACE_DISCOVERY_SECONDS} seconds")
+
+        while True:
+            updated_namespaces: defaultdict = defaultdict(list)
+
+            try:
+                for resource in self.namespace_monitored_resources:
+                    try:
+                        results = Discovery.count_resources(
+                            kind=resource.kind,
+                            api_group=resource.apiGroup,
+                            version=resource.apiVersion
+                        )
+
+                        for namespace_name, count in results.items():
+                            updated_namespaces[namespace_name].append(ResourceCount(
+                                kind=resource.kind,
+                                apiVersion=resource.apiVersion,
+                                apiGroup=resource.apiGroup,
+                                count=count
+                            ))
+
+                    except ResourceAccessForbiddenError as e:
+                        logging.warning(f"Skipping resource {resource.kind} due to insufficient permissions: {e}")
+                    except Exception as e:
+                        logging.exception(f"Unexpected error counting resource {resource.kind}: {e}")
+
+                namespaces_to_publish = []
```

Comment:
I think the namespace publishing should be a part of the discovery, and not in parallel on a separate thread